subsys/dfu/dfu_target/src/dfu_target_suit.c

```diff
@@ -155,5 +188,46 @@ int dfu_target_suit_schedule_update(int img_num)
 int dfu_target_suit_reset(void)
 {
 	stream_buf_bytes = 0;
-	return dfu_target_stream_reset();
+
+	int rc = dfu_target_stream_reset();
+
+	if (rc != 0) {
+		LOG_ERR("dfu_target_stream_reset error %d", rc);
+		return rc;
+	}
+
+	suit_ssf_err_t err = suit_dfu_cleanup();
+
+	stream_flash_in_use = false;
+	image_num = 0;
```

Comment:
I'm wondering if `stream_buf_bytes` should be reset too.

Reply:
It is reset on dfu_target_init so if we want to reset the whole state it should be reset as well. It is done above in this function, but I'll move it here.